app/src/main/java/com/example/tasky/agenda/data/AgendaRepositoryImpl.kt

```diff
@@ -0,0 +1,82 @@
+package com.example.tasky.agenda.data
+
+import android.annotation.SuppressLint
+import com.example.tasky.agenda.data.local.AgendaDao
+import com.example.tasky.agenda.data.mapper.toDomain
+import com.example.tasky.agenda.data.mapper.toEntity
+import com.example.tasky.agenda.data.remote.AgendaService
+import com.example.tasky.agenda.data.util.toCurrentTimeMilli
+import com.example.tasky.agenda.data.util.toEndOfDayLong
+import com.example.tasky.agenda.data.util.toStartOfDayLong
+import com.example.tasky.agenda.domain.AgendaRepository
+import com.example.tasky.agenda.domain.model.AgendaItem
+import com.example.tasky.core.data.Resource
+import com.example.tasky.core.data.remote.safeApiCall
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.merge
+import java.time.LocalDate
+import javax.inject.Inject
+
+@SuppressLint("NewApi")
+class AgendaRepositoryImpl @Inject constructor(
+    private val agendaService: AgendaService,
+    private val agendaDao: AgendaDao
+) : AgendaRepository {
+
+    override fun getAgenda(date: LocalDate): Flow<List<AgendaItem>> = callbackFlow {
```

Comment:
You could replace "callbackFlow" with "flow"
And instead of send, you can use "emit"

Reply:
with `flow` and `emit()` I get this
![image](https://github.com/MelvinKucuk/tasky/assets/42274125/cbe7868c-95d9-4637-ab71-ef44f8eda4c9)